File: summer2021/unit1/fizz-buzz.py
class Solution:
    def fizzBuzz(self, n: int) -> List[str]:
        result = []
        
        for i in range(1, n + 1):
            div_by_5 = False
            div_by_3 = False

            if i % 5 == 0:
                div_by_5 = True
            if i % 3 == 0:
                div_by_3 = True

            if div_by_5 and div_by_3:
                result.append("FizzBuzz")
                continue
            if div_by_3:
                result.append("Fizz")
                continue
            if div_by_5:
                result.append("Buzz")
            else:
                result.append(str(i))
        
        return result
        

File: 2026/week1/fizz_buzz.cpp
class Solution {
public:
    vector<string> fizzBuzz(int n) {
        vector<string> result; 
        for (int i = 1; i < n + 1; i++) {
            bool div_by_3 = i % 3 == 0;
            bool div_by_5 = i % 5 == 0;
            if (div_by_3 && div_by_5) {
                result.push_back("FizzBuzz");
            } else if (div_by_3) {
                result.push_back("Fizz");
            } else if (div_by_5) {
                result.push_back("Buzz");
            } else {
                result.push_back(to_string(i));
            }
        }
        return result;
    }
};
